dapp/src/components/shared/CurrencyBalance/index.tsx

```diff
@@ -76,11 +80,23 @@ export function CurrencyBalance({
         {...balanceTextProps}
       >
         {balance}
-        {withDots && ".."}
+        {withDots && "..."}
       </Box>
       <Box as="span" __css={styles.symbol} {...textProps} {...symbolTextProps}>
         {symbol}
       </Box>
     </Box>
   )
+
+  if (withTooltip) {
+    const tooltipLabel = `${bigIntToUserAmount(BigInt(amount ?? 0), decimals, decimals)} ${symbol}`
```

Comment:
We do not display the correct amount in the tooltip, see below: 

<img width="177" alt="Screenshot 2024-11-05 at 14 49 46" src="https://github.com/user-attachments/assets/09c0b7bc-bf20-4820-aca7-0eebc242b614">

Note that we actually want to calculate the value the same way as for the `balance` variable but in this case `desiredDecimals` will be equal to `decimals` from the currency.

Reply:
> `desiredDecimals` will be equal to `decimals`

It's not necessarily the case. For Bitcoin the `desiredDecimals` is 8. When amount "length" is lower than that it causes the offset you noticed. It's impossible to define it with constant value since it's specific to the amount. Sometimes it's 6, 7, rarely 8.

```ts
const fullPrecisionDecimals = (amount ?? 0).toString().length
console.log(amount, fullPrecisionDecimals)
```

<img width="146" alt="image" src="https://github.com/user-attachments/assets/70bcc87e-0c2b-4194-a726-4efe7ba28398">

I came up with idea to dynamically read the precision per amount. Looks like it's accurate.

<img width="286" alt="image" src="https://github.com/user-attachments/assets/a8c1a9a9-49e7-4918-b0c1-5f5005bd37b4">

Also I transformed the `balance` memoized value to `getBalance` memoized function. It allowed me to get exact the same balance in multiple places without repetition of the code. Performance-wise it's also covered by memoization.

Ref commit: 0fe9988a

Reply:
If the precision is critical and must be as accurate as possible then we use the method - `fixedPointNumberToString`. Inside the `bigIntToUserAmount` the `fixedPointDecimals` is subtracted by `desiredDecimals` which results as **0** (8 - 8). It leads to unexpected value offsets like **0.12200000** instead of expected **1.22000000**.

Ref commit: 9d13616c